From: Bastian Blank <waldi@debian.org>
Description: skip non-existing devices in inittab
Bug-Debian: http://bugs.debian.org/541115
Forwarded: no

This patch causes init silently skip running processes from inittab if
the terminal name is specified but the corresponding device file does not
exist.

--- a/init/init.c
+++ b/init/init.c
@@ -563,6 +563,8 @@ static void run_actions(int action_type)
 	for (a = init_action_list; a; a = a->next) {
 		if (!(a->action_type & action_type))
 			continue;
+		if (a->terminal[0] && access(a->terminal, R_OK | W_OK))
+			continue;
 
 		if (a->action_type & (SYSINIT | WAIT | ONCE | CTRLALTDEL | SHUTDOWN)) {
 			pid_t pid = run(a);
